### DISCUSSION TITLE
Add sequential ID to logs collected by fluent bit

### DISCUSSION THREAD

**Discussion by ghost** [fluent/fluent-bit#9654] [Q&A]
I have some usecases where multiple logs are printed at quick succession and timestamp remains exactly the same.

I ingest the logs into opensearch and when querying, these logs don't come in same order.

Is it possible to remediate this? Is there a way to generate a sequential number for each log from the same `tail` source?

**Comment by patrick-stephens** [comment]
you could use a lua filter or similar to handle it, bit of script to cope with incrementing the timer or resetting it on new timestamp
